agent/service-message-maven-extension/src/main/java/nu/studer/teamcity/buildscan/agent/servicemessage/BuildScanServiceMessageMavenExtension.java

```diff
@@ -32,9 +33,16 @@ public void configure(GradleEnterpriseApi api, MavenSession session) {
             logger.debug("Registering listener capturing build scan link");
             BuildScanServiceMessageSender.register(api.getBuildScan());
             logger.debug("Finished registering listener capturing build scan link");
+            configureScans(api.getBuildScan());
         }
     }
 
+    private void configureScans(BuildScanApi api) {
+        logger.debug("Configuring build scan to be published always");
+        api.publishAlways();
+        api.value("CI auto injection", "TeamCity");
+    }
+
```

Comment:
Unfortunately, this is not the right place. Please make the additions here:
https://github.com/etiennestuder/teamcity-build-scan-plugin/blob/90860abd69edcbb476c8dfc0513ac1f624b2e50e/agent/src/main/java/nu/studer/teamcity/buildscan/agent/BuildScanServiceMessageInjector.java#L244-L245